clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/CreateGoogleHttpLoadBalancerAtomicOperation.groovy

```diff
@@ -74,133 +74,132 @@ class CreateGoogleHttpLoadBalancerAtomicOperation  implements AtomicOperation<De
-    }
 
-    def hostRules = new ArrayList<HostRule>();
-    for (hostRule in description.hostRules) {
-      hostRules.add(
-          hosts: hostRule.hosts,
-          pathMatcher: hostRule.pathMatcher
+      task.updateStatus BASE_PHASE, "Creating health check $healthCheckName..."
+      HttpHealthCheck newHealthCheck = new HttpHealthCheck(
+          name: healthCheckName,
+          port: healthCheck.port,
+          requestPath: healthCheck.requestPath,
+          checkIntervalSec: healthCheck.checkIntervalSec,
+          healthyThreshold: healthCheck.healthyThreshold,
+          unhealthyThreshold: healthCheck.unhealthyThreshold,
+          timeoutSec: healthCheck.timeoutSec,
       )
+      def insertHealthCheckOperation = compute.httpHealthChecks().insert(project, newHealthCheck).execute()
+      googleOperationPoller.waitForGlobalOperation(compute, project, insertHealthCheckOperation.getName(),
+          null, task, "health check " + healthCheckName, BASE_PHASE)
     }
 
-    def urlMapName = String.format("%s-%s-%d", description.loadBalancerName, URL_MAP_NAME_PREFIX, System.currentTimeMillis())
-
-    task.updateStatus BASE_PHASE, "Creating URL map $urlMapName..."
-
-    def urlMap = new UrlMap(
-        name: urlMapName,
-        pathMatchers: pathMatchers,
-        hostRules: hostRules,
-        defaultService: backendServiceUrl
-    )
-
-    // Before building the url map we must check and wait until the backend service is built.
-    googleOperationPoller.waitForGlobalOperation(compute, project, backendServiceOperation.getName(),
-        null, task, "backend service " + GCEUtil.getLocalName(backendServiceUrl), BASE_PHASE)
-
-    def urlMapOperation = compute.urlMaps().insert(project, urlMap).execute()
-    def urlMapUrl = urlMapOperation.getTargetLink()
-
-    // Make target HTTP proxy.
-    def targetHttpProxyName = String.format("%s-%s-%d", description.loadBalancerName, TARGET_HTTP_PROXY_NAME_PREFIX, System.currentTimeMillis())
-
-    task.updateStatus BASE_PHASE, "Creating target HTTP proxy $targetHttpProxyName..."
+    List<BackendService> existingServices = compute.backendServices().list(project).execute().getItems()
+    backendServicesFromDescription.each { GoogleBackendService backendService ->
+      String backendServiceName = "$httpLoadBalancerName-$BACKEND_SERVICE_NAME_PREFIX-$backendService.name"
 
-    def targetHttpProxy = new TargetHttpProxy(
-        name: targetHttpProxyName,
-        urlMap: urlMapUrl
-    )
-
-    // Before building the target http proxy we must check and wait until the url map is built.
-    googleOperationPoller.waitForGlobalOperation(compute, project, urlMapOperation.getName(),
-        null, task, "url map " + GCEUtil.getLocalName(urlMapUrl), BASE_PHASE)
+      def existingService = existingServices.findAll { it.name == backendServiceName }
+      if (existingService) {
+        throw new GoogleOperationException("There is already a backend service named ${backendService.name} " +
```

Comment:
Agreed, but it's well reported. I'll address this in the hop from create to upsert.